cli/azd/pkg/azdo/service_connection.go

```diff
@@ -132,36 +145,46 @@ func CreateServiceConnection(
 
 	err = authorizeServiceConnectionToAllPipelines(ctx, projectId, endpoint, connection)
 	if err != nil {
-		return fmt.Errorf("authorizing service connection: %w", err)
+		return nil, fmt.Errorf("authorizing service connection: %w", err)
 	}
 
-	return nil
+	return endpoint, nil
+}
+
+func ListTypes(
+	ctx context.Context,
+	connection *azuredevops.Connection,
+	projectId string) (*[]serviceendpoint.ServiceEndpointType, error) {
+
+	client, err := serviceendpoint.NewClient(ctx, connection)
+	if err != nil {
+		return nil, fmt.Errorf("creating new azdo client: %w", err)
+	}
+
+	return client.GetServiceEndpointTypes(ctx, serviceendpoint.GetServiceEndpointTypesArgs{})
 }
 
 // creates input parameter needed to create the azure rm service connection
 func createAzureRMServiceEndPointArgs(
-	ctx context.Context,
 	projectId *string,
+	projectName *string,
 	credentials *azcli.AzureCredentials,
 ) (serviceendpoint.CreateServiceEndpointArgs, error) {
-	endpointType := "azurerm"
-	endpointOwner := "library"
-	endpointUrl := "https://management.azure.com/"
-	endpointName := ServiceConnectionName
-	endpointIsShared := false
-	endpointScheme := "ServicePrincipal"
-
+	endpointScheme := "WorkloadIdentityFederation"
 	endpointAuthorizationParameters := map[string]string{
-		"serviceprincipalid":  credentials.ClientId,
-		"serviceprincipalkey": credentials.ClientSecret,
-		"authenticationType":  "spnKey",
-		"tenantid":            credentials.TenantId,
+		"serviceprincipalid": credentials.ClientId,
+		"tenantid":           credentials.TenantId,
+	}
+	if credentials.ClientSecret != "" {
+		endpointAuthorizationParameters["serviceprincipalkey"] = credentials.ClientSecret
+		endpointAuthorizationParameters["authenticationType"] = "spnKey"
+		endpointScheme = "ServicePrincipal"
 	}
 
 	endpointData := map[string]string{
 		"environment":      CloudEnvironment,
 		"subscriptionId":   credentials.SubscriptionId,
-		"subscriptionName": "azure subscription",
+		"subscriptionName": "azure subscription", // fix? to sub name?
```

Comment:
Did you want to do something additional here, or file a follow up issue?

Reply:
removed the comment. 
I was not sure about this value was going to be a problem for OIDC, but it did not matter